build-system/tasks/helpers.js

```diff
@@ -220,12 +220,14 @@ function appendToCompiledFile(srcFilename, destFilePath) {
   }
 }
 
-function toEsmName(name) {
-  return name.replace(/\.js$/, '.mjs');
-}
-
-function maybeToEsmName(name) {
-  return argv.esm ? toEsmName(name) : name;
+function getMinifiedName(name) {
+  if (argv.sxg) {
+    return name.endsWith('.sxg.js') ? name : name.replace(/\.js$/, '.sxg.js');
+  }
```

Comment:
Maybe rewrite this clause to `if (argv.sxg && !names.endsWith('.sxg.js'))` and remove the ternary operator so that you're not returning `name` from two different branches?